uploader/spreadsheet.py

```diff
@@ -0,0 +1,570 @@
+import logging
+from typing import List
+
+import pandas as pd
+from pandas import ExcelFile
+
+from institution.models import CofkCollectInstitution
+from location.models import CofkCollectLocation
+from manifestation.models import CofkCollectManifestation
+from person.models import CofkCollectPerson
+from uploader import OpenpyxlReaderWOFormatting
+from uploader.constants import mandatory_sheets
+from uploader.models import CofkCollectUpload, Iso639LanguageCode
+from uploader.validation import validate_work, validate_manifestation, CofkExcelFileError
+from work.models import CofkCollectAuthorOfWork, CofkCollectAddresseeOfWork, CofkCollectWork, \
+    CofkCollectPersonMentionedInWork, CofkCollectLanguageOfWork, CofkCollectWorkResource
+
+log = logging.getLogger(__name__)
+
+
+class CofkRepositories:
+    def __init__(self, upload: CofkCollectUpload, sheet_data: pd.DataFrame, limit=None):
+        self.upload = upload
+
+        self.__institution_id = None
+        self.sheet = sheet_data
+        self.__repository_data = {}
+        limit = limit if limit else len(self.sheet.index)
+        self.ids = []
+
+        # Process each row in turn, using a dict comprehension to filter out empty values
+        for i in range(1, limit):
+            self.process_repository({k: v for k, v in self.sheet.iloc[i].to_dict().items() if v is not None})
```

Comment:
class of (`CofkRepositories`, `CofkLocations`, `CofkPeople`, `CofkManifestations`) are very similar, you may want to simplify or make it more general in the future